### PR TITLE
kdump is currently not compatible with Systemd Boot

### PR DESCRIPTION
**Pull request by schubi2**
## Problem

mkdumprd produces an error while installation if kdump is used together with systemd-boot.

## Solution

Do not propose Kdump if systemd-boot is used for bootloader.
If the user still insists on using Kdump he will get a warning but can continue.

It seems that the Kdump scripts have problems with systemd-boot. See
https://bugzilla.opensuse.org/show_bug.cgi?id=1226676

We are aware that this PR is temporary only until Kdump fixed. But meanwhile we should inform the user
that there could be a problem here. Currently we are already tell the user in YaST that systemd-boot
is in the evaluation state if he is selecting it for bootloader.

## Testing
- Tested manually

### CODE REVIEW

**Comment by jiribohac** [comment]
Why don't we fix the kdump/systemd-boot incompatibility instead?
Could you share more about the problem, I don't know anything about it? bsc?

**Comment by jiribohac** [comment]
Please report the problem in bugzilla and let's work to fix this in kdump or systemd-boot or both.
I think adding such workaround into yast2-kdump is not a good idea We need to fix the problem. If it's not possible, we could also easily prevent concurrent installation of kdump and systemd-boot by simply adding a Conflicts: into one of the packages' spec file.

**Comment by schubi2** [comment]
> Please report the problem in bugzilla and let's work to fix this in kdump or systemd-boot or both. I think adding such workaround into yast2-kdump is not a good idea We need to fix the problem. If it's not possible, we could also easily prevent concurrent installation of kdump and systemd-boot by simply adding a Conflicts: into one of the packages' spec file.

Yes, I will create a bugzilla entry but fixing Kdump is not so easy because currently the Kdump scripts does not support systemd-boot. Solving it via package dependencies would come to the same solution as this PR has already.
We are at the beginning of systemd-boot. Currently we are collecting more or less "problems" in TW. So this fix is temporary only and should only inform the user that there could be a problem but it would be not fair if we are not giving this information to the user. Our aim is that other users are "playing" with systemd-boot in TW :-)

P.S.: Do have any idea why Rubocop is failing ?

File: src/clients/kdump_auto.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (jreidinger) [inline src/clients/kdump_auto.rb:1]: ah, I would not not add that automatically to files as it is potentially dangerous ( ruby will raise exception if code manipulate with frozen string literal and sadly old ycp code doing it quite lot.

File: src/modules/Kdump.rb [hunk 2]
```diff
@@ -897,19 +901,26 @@ def proposal_warning
...
+        # TRANSLATORS: warning message in installation proposal. Do not translate %{requested} and
+        # %{available} - they are replaced with actual sizes later.
+        warning_string = format(_(
+          "Warning! There might not be enough free space to have kdump enabled. " \
+          "%{required} required for saving a kernel dump, but only %{available} are available."
+        ), required: String.FormatSizeWithPrecision(requested_space, 2, true), available: String.FormatSizeWithPrecision(free_space, 2, true))
+      end
+
+      if Bootloader.getLoaderType == "systemd-boot" && @add_crashkernel_param
+        warning_string += "<br>" if !warning_string.empty?
```
Comment (jreidinger) [inline src/modules/Kdump.rb:916]: minor tip: you can use `unless` instead of `if !`

Comment (schubi2) [reply src/modules/Kdump.rb]: There has been some more :-)

File: src/clients/kdump_auto.rb (hunk 1 above)
Comment (schubi2) [reply src/clients/kdump_auto.rb]: I have removed it

**Review by jreidinger — APPROVED** [review]

**Pull request merged**